### PR TITLE
[Notifications] Mask notification secret params on submit_workflow [1.7.x]

### PR DESCRIPTION
**Pull request by roei3000b**
We need to mask  notification secret params on submit_workflow so we can store the notification secret params as k8s secret.

https://iguazio.atlassian.net/browse/ML-8029

### CODE REVIEW

File: server/api/api/utils.py [hunk 1]
```diff
@@ -223,22 +223,58 @@ def mask_notification_params_on_task(
...
     notifications = task.get("spec", {}).get("notifications", [])
-    masked_notifications = []
     if notifications:
-        for notification in notifications:
-            notification_object = mlrun.model.Notification.from_dict(notification)
-            masked_notifications.append(
-                mask_op(project, run_uid, notification_object).to_dict()
-            )
-        task.setdefault("spec", {})["notifications"] = masked_notifications
+        task.setdefault("spec", {})["notifications"] = _mask_notifications_params(
```
Comment (theSaarco) [inline server/api/api/utils.py]: Looks like in this flow you're missing a `to_dict()` on the masked notifications - before your change the code would do `mask_op(...).to_dict()`, which is something you don't do now.

**Review by theSaarco — CHANGES_REQUESTED** [review]
Looks fine, but there's a missing conversion in one flow.

File: server/api/api/utils.py (hunk 1 above)
Comment (roei3000b) [reply server/api/api/utils.py]: You right, thanks 👍

**Review by liranbg — APPROVED** [review]

**Review by theSaarco — APPROVED** [review]
LGTM.

File: server/api/crud/workflows.py [hunk 2]
```diff
@@ -233,6 +240,11 @@ def run(
             run_name=runner.metadata.name,
             load_only=load_only,
         )
+        # We want to store the secret params as k8s secret, so later we can access them with the project internal secret
+        # key that was created.
+        server.api.api.utils.mask_notification_params_on_task_object(
+            run_spec, server.api.constants.MaskOperations.CONCEAL
+        )
```
Comment (alonmr) [inline server/api/crud/workflows.py:247]: there is some common code here together with node selector enrichment with the schedule flow can be merged to method

**Pull request merged**